cli/dotfiles.go

```diff
@@ -140,6 +146,23 @@ func (r *RootCmd) dotfiles() *clibase.Cmd {
 				_, _ = fmt.Fprintln(inv.Stdout, cliui.DefaultStyles.Error.Render("Failed to update repo, continuing..."))
 			}
 
+			if dotfilesExists && gitbranch != "" {
+				// If the repo exists and the git-branch is specified, we need to check out the branch. We do this after
+				// git pull to make sure the branch was pulled down locally. If we do this before the pull, we could be
+				// trying to checkout a branch that does not yet exist locally and get a git error.
+				_, _ = fmt.Fprintf(inv.Stdout, "Dotfiles git branch %q specified\n", gitbranch)
+				err := ensureCorrectGitBranch(inv, ensureCorrectGitBranchParams{
+					repoDir:       dotfilesDir,
+					gitSSHCommand: gitsshCmd,
+					gitBranch:     gitbranch,
+				})
+				if err != nil {
+					// Do not block on this error, just log it and continue
+					_, _ = fmt.Fprintln(inv.Stdout,
+						cliui.DefaultStyles.Error.Render(fmt.Sprintf("Failed to use branch %q (%s), continuing...", err.Error(), gitbranch)))
```

Comment:
Would it make sense to use a non-zero exit code if this happens? Otherwise the user might expect one thing but receive another due to the silent failure. (Not saying we can't continue from here, but set a `branchFailed = true` and then exit 1 at the end.

Reply:
That is an idea. I am not sure, I was copying the existing behavior.

See this: 
https://github.com/coder/coder/blob/e1d7dc38fcf323114a2b1ab166b71be0d75d43ca/cli/dotfiles.go#L142-L142

If the `git pull --ff-only` fails then we continue, even though we are now operating in a stale dotfiles. I was maintaining that behavior of continuing.

If we decide to exit code 1, we should go back and also change the previous behavior.